File: RMI/Ride.java
// Ride.java
import java.io.Serializable;
import java.time.LocalDateTime;

class Ride implements Serializable {
    private String rideId;
    private String riderName;
    private String driverName;
    private String pickup;
    private String destination;
    private String status; // "REQUESTED", "ACCEPTED", "IN_PROGRESS", "COMPLETED", "CANCELLED"
    private LocalDateTime bookingTime;
    private double fare;
    
    public Ride(String rideId, String riderName, String pickup, String destination) {
        this.rideId = rideId;
        this.riderName = riderName;
        this.pickup = pickup;
        this.destination = destination;
        this.status = "REQUESTED";
        this.bookingTime = LocalDateTime.now();
    }
    
    // Getters and setters
    public String getRideId() { return rideId; }
    public String getRiderName() { return riderName; }
    public String getDriverName() { return driverName; }
    public void setDriverName(String driverName) { this.driverName = driverName; }
    public String getPickup() { return pickup; }
    public String getDestination() { return destination; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public LocalDateTime getBookingTime() { return bookingTime; }
    public double getFare() { return fare; }
    public void setFare(double fare) { this.fare = fare; }
}

File: WebSocket/ride.py
from datetime import datetime

class Ride:
    def __init__(self, ride_id, rider_name, pickup, destination):
        self.ride_id = ride_id
        self.rider_name = rider_name
        self.driver_name = None
        self.pickup = pickup
        self.destination = destination
        self.status = "REQUESTED"  # "REQUESTED", "ACCEPTED", "IN_PROGRESS", "COMPLETED", "CANCELLED"
        self.booking_time = datetime.now().isoformat()
        self.fare = 0.0
    
    def to_dict(self):
        return {
            'ride_id': self.ride_id,
            'rider_name': self.rider_name,
            'driver_name': self.driver_name,
            'pickup': self.pickup,
            'destination': self.destination,
            'status': self.status,
            'booking_time': self.booking_time,
            'fare': self.fare
        }
    
    @classmethod
    def from_dict(cls, data):
        ride = cls(data['ride_id'], data['rider_name'], 
                  data['pickup'], data['destination'])
        ride.driver_name = data.get('driver_name')
        ride.status = data['status']
        ride.booking_time = data['booking_time']
        ride.fare = data['fare']
        return ride
    
    def __str__(self):
        return f"Ride {self.ride_id}: {self.rider_name} from {self.pickup} to {self.destination} - ₹{self.fare} ({self.status})"